tests/trusted-client-tests.js

```diff
@@ -105,6 +106,71 @@ describe('TrustedClient', function() {
       });
     });
 
+    it('does not send body for GET when json is true', function() {
+      return client.request(uri, { method: 'GET', json: true })
+        .then(({ statusCode, body }) => {
+          expect(statusCode).to.equal(200);
+          expect(body).to.deep.equal(defaultResponse);
+        });
+    });
+
+    it('does send body for GET when told to', function() {
+      const requestBody = { thisis: 'dumb' };
+      return client.request(uri, { method: 'GET', json: requestBody })
+        .then(({ statusCode, body }) => {
+          expect(statusCode).to.equal(200);
+          expect(body).to.deep.equal(requestBody);
+        });
+    });
+
+    it('returns no error when statusCode is below client errorStatus', function() {
+      client = TrustedClient({
+        keyId: 'test',
+        key: privateKey,
+        errorStatus: 300
+      });
+      return client.request(uri, { method: 'GET', headers: { 'x-expected-status-code': 200 } })
+        .then(({ statusCode, body }) => {
+          expect(statusCode).to.equal(200);
+          expect(body).to.deep.equal(defaultResponse);
+        });
+    });
+
+    it('returns error when statusCode is above client errorStatus', function() {
+      client = TrustedClient({
+        keyId: 'test',
+        key: privateKey,
+        errorStatus: 300
+      });
+      return client.request(uri, { method: 'GET', headers: { 'x-expected-status-code': 300 } })
+        .then(() => {
+          fail('Expected error, got success');
+        })
+        .catch((err) => {
+          expect(err.statusCode).to.equal(300);
+          expect(err.body).to.equal(JSON.stringify(defaultResponse));
+        });
+    });
+
+    it('returns no error when statusCode is below request errorStatus', function() {
+      return client.request(uri, { method: 'GET', errorStatus: 300, headers: { 'x-expected-status-code': 200 } })
+        .then(({ statusCode, body }) => {
+          expect(statusCode).to.equal(200);
+          expect(body).to.deep.equal(defaultResponse);
+        });
+    });
+
+    it('returns error when statusCode is above request errorStatus', function() {
+      return client.request(uri, { method: 'GET', errorStatus: 300, headers: { 'x-expected-status-code': 300 } })
+        .then(() => {
+          fail('Expected error, got success');
```

Comment:
You may need to use the `done` callback for these kinds of tests because the error messaging may be troublesome
i.e.
```js
.then(() => done(new Error('Expected error, got success')))
.catch(err => ...)
```